gremlin-go/driver/graphTraversalSource.go

```diff
@@ -19,5 +19,132 @@ under the License.
+	bytecode            *bytecode
+	remoteConnection    *DriverRemoteConnection
+}
+
+// NewGraphTraversalSource creates a graph traversal source, the primary DSL of the Gremlin traversal machine
+func NewGraphTraversalSource(graph *Graph, traversalStrategies *TraversalStrategies, bytecode *bytecode) *GraphTraversalSource {
+	return &GraphTraversalSource{graph: graph, traversalStrategies: traversalStrategies, bytecode: newBytecode(bytecode)}
+}
+
+// GetBytecode gets the traversal bytecode associated with this graph traversal source
+func (gts *GraphTraversalSource) GetBytecode() *bytecode {
+	return gts.bytecode
+}
+
+// GetTraversalStrategies gets the graph traversal strategies associated with this graph traversal source
+func (gts *GraphTraversalSource) GetTraversalStrategies() *TraversalStrategies {
+	return gts.traversalStrategies
+}
+
+func (gts *GraphTraversalSource) clone() *GraphTraversalSource {
+	return NewGraphTraversalSource(gts.graph, gts.traversalStrategies, newBytecode(gts.bytecode))
+}
+
+// WithBulk allows for control of bulking operations
+func (gts *GraphTraversalSource) WithBulk(args ...interface{}) *GraphTraversalSource {
+	source := gts.clone()
+	err := source.bytecode.addSource("withBulk", args)
+	if err != nil {
+		return nil
+	}
+	return source
+}
+
+func (gts *GraphTraversalSource) WithPath(args ...interface{}) *GraphTraversalSource {
+	source := gts.clone()
+	source.bytecode.addSource("withPath", args)
+	return source
+}
+
+// WithSack adds a sack to be used throughout the life of a spawned Traversal
+func (gts *GraphTraversalSource) WithSack(args ...interface{}) *GraphTraversalSource {
+	source := gts.clone()
+	source.bytecode.addSource("withSack", args)
+	return source
+}
+
+// WithSideEffect adds a side effect to be used throughout the life of a spawned Traversal
+func (gts *GraphTraversalSource) WithSideEffect(args ...interface{}) *GraphTraversalSource {
+	source := gts.clone()
+	source.bytecode.addSource("withSideEffect", args)
+	return source
+}
+
+// WithStrategies adds an arbitrary collection of TraversalStrategies instances to the traversal source
+func (gts *GraphTraversalSource) WithStrategies(args ...interface{}) *GraphTraversalSource {
+	source := gts.clone()
+	source.bytecode.addSource("withStrategies", args)
+	return source
+}
+
+// WithoutStrategies removes an arbitrary collection of TraversalStrategies instances to the traversal source
+func (gts *GraphTraversalSource) WithoutStrategies(args ...interface{}) *GraphTraversalSource {
+	source := gts.clone()
+	source.bytecode.addSource("withoutStrategies", args)
+	return source
+}
+
+// With provides a configuration to a traversal in the form of a key value pair
+// TODO: Add to this when implementing traversal strategies
+func (gts *GraphTraversalSource) With(key interface{}, value interface{}) *GraphTraversalSource {
+	source := gts.clone()
+	source.bytecode.addSource("withStrategies", key, value)
+	return source
+}
```

Comment:
I suggest removing this until we have a proper implementation for our traversal strategies.
```suggestion
//func (gts *GraphTraversalSource) With(key interface{}, value interface{}) *GraphTraversalSource {
//	source := gts.clone()
//	source.bytecode.addSource("withStrategies", key, value)
//	return source
//}
```